### PR TITLE
#53 Create Application Review Queue endpoint

### LINKED ISSUE #53: Application Review Queue
# Create Application Review Queue
We need a fun queue system for application reviews! This issue is to basically write the query to pull the *top* of the queue.

### Create protected `reviewRequestRouter`
- Add constants `REQUIRED_REVIEWS` and `REVIEW_TIMEOUT`
- Add boolean `completed` column to `reviews` table
- Input should be the `user.id` of the reviewer
- Create a query to fetch the first (maybe using `findFirst()`?) application from the `application` table that requires review with the following parameters:
  -  **select** all applications
  - **where** # reviews < `REQUIRED_REVIEWS` **and**  (pending review **or** (in review **and** current time - review creation time > `REVIEW_TIMEOUT`))
  - **order by** # reviews
- Update the status of the application to "IN_REVIEW"
- If review exists and timeout exceeded, remove the review first
- Create new review associated with the application

### Notes
- Reviews table might have to be joined or filtered in order to pick an in-review application that has timed out

### PR DESCRIPTION
**Pull request by LordExodius**
closes #53 

# Quick Overview of Changes

- Added `reviewRequestRouter`
- Created SQL queries to fetch applications for review

# Checklist

- [x] Checked all uses of changed component(s)/function(s)
- [ ] If any changes were made to our backend services, at least one (happy path) unit test was added OR a `TODO` comment was added to create one.
- [ ] Check that CI is passing.
- [ ] If no (re-)reviews after 1-2 days, ping the web leads on the discord to remind them to review the PR.

@basokant @ArsalaanAli

### CODE REVIEW

**Comment by hunterchen7** [comment]
don't merge, doesn't work 👎🏼

File: src/server/api/root.ts [hunk 1]
```diff
@@ -15,7 +15,7 @@ export const appRouter = createTRPCRouter({
   application: applicationRouter,
   auth: authRouter,
   preregistration: preregistrationRouter,
-  review: reviewRouter,
```
Comment (basokant) [inline src/server/api/root.ts:18]: don't replace the review router (I'm guessing this happened when resolving conflicts), either don't create a new router and add ur mutation to the existing reviewRouter (I'd recommend this), or create a new router here for ur reviewRequestRouter

**Review by basokant — CHANGES_REQUESTED** [review]
does not look good to me :((

File: src/server/api/root.ts (hunk 1 above)
Comment (basokant) [reply src/server/api/root.ts:18]: this is what's causing build errors lol

**Review by basokant — APPROVED** [review]
lgtm

**Pull request merged**